### ISSUE TITLE
Can´t create entity with two HasOne attributes pointing to same entity

### ISSUE THREAD

**Issue by milosloub** [json-api-dotnet/JsonApiDotNetCore#322]
I have two entities Client and ClientRelation. ClientRelation only stores two Client foreign keys and their relation 

ClientRelation looks like this

```
    public class ClientRelation : Identifiable
    {
        [HasOne("client", Link.None)]
        public virtual Client Client { get; set; }
        public int ClientId { get; set; }

        [HasOne("relatedClient", Link.None)]
        public virtual Client RelatedClient { get; set; }
        public int RelatedClientId { get; set; }

        [Attr("relationship")]
        public RelationshipEnum Relationship { get; set; }
    }
```
After upgrade to version 2.3.2 or 2.3.3, POST on this entity throws exception:

> System.Data.SqlClient.SqlException: Cannot insert explicit value for identity column in table 'Clients' when IDENTITY_INSERT is set to OFF.

Post request looks like this:
```
{
    "data": {
        "attributes": {
            "relationship":1
        },
        "relationships": {
        	"client": {
                "data": {
                    "type": "clients",
                    "id": "1"
                }
            },
    		"relatedClient": {
                "data": {
                    "type": "clients",
                    "id": "2"
                }
            },
        },
        "type": "client-relations"
    }
}
```

In version 2.3.1 everything is OK. I think that is somehow caused by new method AttachHasOnePointers() in DefaultEntityRepository. Do you know, what could be a problem?

**Comment by milosloub** [comment]
Thank you ! I'll try it when new version will be out. Fix looks great 👍

**Comment by jaredcnance** [comment]
I am bundling it with some fixes for json:api v1.1 operations. This shouldn't impact you unless you're using bulk operations but the version has been bumped to v2.4.0-beta1

**Comment by milosloub** [comment]
I tried 2.4.0-beta1 (I don't use operations ) and now everythink is ok. Thank you 👍
